### PR TITLE
SEARCH-1513 ADRs for combining the search codebases.

### PR DESCRIPTION
**Pull request by tpage-alfresco**

### CODE REVIEW

File: alfresco-search/doc/architecture/decisions/0004-community-mirror.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+A ticket has been raised to track this [[1]]. A script has been written to do the initial codebase merge and
+mirroring [[2]].
+
+We will rewrite the whole history of the SearchServices repository using the mirroring command (the only change will be
+the commit ids).
+
+Any changes within a module called alfresco-search-parent will be mirrored. Any changes outside this will not be
+mirrored. 
+
+!!! TODO: Check if the community can build the submodule without access to the parent pom file !!!
```
Comment (tpage-alfresco) [inline alfresco-search/doc/architecture/decisions/0004-community-mirror.md]: This is not possible. They get the following error:
```
Non-resolvable parent POM for org.alfresco:alfresco-search-parent:1.2.0-IE1.0.1-SNAPSHOT: Could not find artifact org.alfresco:alfresco-search-and-insight-parent:pom:8 in alfresco-internal (https://artifacts.alfresco.com/nexus/content/groups/internal/) and 'parent.relativePath' points at wrong local POM @ line 4, column 13 -> [Help 2]
```

Comment (tpage-alfresco) [reply alfresco-search/doc/architecture/decisions/0004-community-mirror.md]: We can use `git filter-branch` rather than `git subtree` to resolve this.

**Comment by michaelsuzukisagi** [comment]
LGTM

**Comment by michaelsuzukisagi** [comment]
LGTM

**Pull request merged**